src/custom/state/unsupportedTokens/hooks.ts

```diff
@@ -0,0 +1,29 @@
+import { useCallback } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { ChainId } from '@uniswap/sdk'
+import { AppDispatch, AppState } from 'state'
+import { UnsupportedTokensState } from 'state/unsupportedTokens/reducer'
+import { addUnsupportedToken, removeUnsupportedToken, UnsupportedTokenParams } from './actions'
+
+export function useDynamicallyUnsupportedTokens(chainId?: ChainId) {
```

Comment:
Do we need to use the word `Dynamically`, is it becuase theres another hook `useUnsupportedTokens`. 
Another alternative is call it `useGpUnsupportedTokens`

Then, I would create a general hook `useIsUnssupportedToken` that return the predicate function and uses `gpUnsupportedTokens` and `unsupportedTokens` one

Reply:
you need to use the errorType if present, to see if it's one error you can handle

![image](https://user-images.githubusercontent.com/2352112/117953134-223ce280-b316-11eb-9070-847555dab1e4.png)

Reply:
i hate naming functions 😓 i just wait for these comments to rename them from your comments 😂